packages/tx/src/types.ts

```diff
@@ -93,20 +95,98 @@ export function isAccessList(input: AccessListBytes | AccessList): input is Acce
 
 /**
  * Encompassing type for all transaction types.
- *
- * Note that this also includes legacy txs which are
- * referenced as {@link Transaction} for compatibility reasons.
  */
-export type TypedTransaction =
-  | Transaction
-  | AccessListEIP2930Transaction
-  | FeeMarketEIP1559Transaction
-  | BlobEIP4844Transaction
+export enum TransactionType {
+  Legacy = 0,
+  AccessListEIP2930 = 1,
+  FeeMarketEIP1559 = 2,
+  BlobEIP4844 = 3,
+}
+
+export interface Transaction {
+  [TransactionType.Legacy]: LegacyTransaction
+  [TransactionType.FeeMarketEIP1559]: FeeMarketEIP1559Transaction
+  [TransactionType.AccessListEIP2930]: AccessListEIP2930Transaction
+  [TransactionType.BlobEIP4844]: BlobEIP4844Transaction
+}
+
+export type TypedTransaction = Transaction[TransactionType]
+
+export function isLegacyTx(tx: TypedTransaction): tx is LegacyTransaction {
+  return tx.type === TransactionType.Legacy
+}
+
+export function isAccessListEIP2930Tx(tx: TypedTransaction): tx is AccessListEIP2930Transaction {
+  return tx.type === TransactionType.AccessListEIP2930
+}
+
+export function isFeeMarketEIP1559Tx(tx: TypedTransaction): tx is FeeMarketEIP1559Transaction {
+  return tx.type === TransactionType.FeeMarketEIP1559
+}
+
+export function isBlobEIP4844Tx(tx: TypedTransaction): tx is BlobEIP4844Transaction {
+  return tx.type === TransactionType.BlobEIP4844
+}
+
+export interface TransactionInterface<T extends TransactionType> {
+  supports(capability: Capability): boolean
+  type: number
+  validate(): boolean
+  validate(stringError: false): boolean
+  validate(stringError: true): string[]
+  validate(stringError: boolean): boolean | string[]
```

Comment:
Ah, we should urgently have a look there and see that we align this typing to a single version, pretty sure (not 100% though, so this would need some confirmation) these overload-typings have been introduced for some backwards-compatibility reasons.